convert_test.go

```diff
@@ -0,0 +1,70 @@
+package cyclonedx
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_componentConverter_convertEvidence(t *testing.T) {
+	t.Run("spec 1.2 and lower", func(t *testing.T) {
+		convert := componentConverter(SpecVersion1_2)
+
+		comp := Component{
+			Evidence: &Evidence{},
+		}
+
+		convert(&comp)
+
+		assert.Nil(t, comp.Evidence)
+	})
+
+	t.Run("spec 1.4 and lower", func(t *testing.T) {
+		convert := componentConverter(SpecVersion1_4)
+
+		comp := Component{
+			Evidence: &Evidence{
+				Identity:    &EvidenceIdentity{},
+				Occurrences: &[]EvidenceOccurrence{},
+				Callstack:   &Callstack{},
+				Copyright:   &[]Copyright{{Text: "foo"}},
+			},
+		}
+
+		convert(&comp)
+
+		assert.Nil(t, comp.Evidence.Identity)
+		assert.Nil(t, comp.Evidence.Occurrences)
+		assert.Nil(t, comp.Evidence.Callstack)
+		assert.NotNil(t, comp.Evidence.Copyright)
+	})
+
+	t.Run("spec 1.5 and lower", func(t *testing.T) {
+		convert := componentConverter(SpecVersion1_5)
+
+		comp := Component{
+			Evidence: &Evidence{
+				Occurrences: &[]EvidenceOccurrence{
+					{
+						BOMRef:            "foo",
+						Location:          "bar",
+						LineNumber:        42,
+						Offset:            42,
```

Comment:
What if the offset should indeed be `"0"`? We'd have to remove the `omitempty` on the `json` label, but that would break conversion to lower spec versions. Ideally `LineNumber` and `Offset` would be pointers to integers, but they would make it awkward for consumers to create this type. What do you think?

Reply:
@nscuro Since my last comment, I have changed fields `Line` and `Offset` to be `*int`, in line with some other types that have a similar shape.

Reply:
> Ideally LineNumber and Offset would be pointers to integers, but they would make it awkward for consumers to create this type. What do you think?

Agree that it's awkward, in fact I really dislike that we have to do this (#21). But I haven't been able to find a better solution yet. So yeah, using pointers is the way to go for now, thanks for making that adjustment!